### ISSUE TITLE
Horizontal display of the vertical photo taken with the phone.

### ISSUE THREAD

**Issue by gazlag** [wkhtmltopdf/wkhtmltopdf#4846]
**wkhtmltopdf version(s) affected**: 0.12.6

**OS information**
Centos 7

**Description**
The picture that appears vertically in the html page appears horizontal in the pdf created by wkhtmltopdf.
Note: Pictures taken vertically by phone.  (Ios / android)Pictures taken horizontally look normal.

**How to reproduce**
<section class="page photo-page">
     <img src="images/125698_05.jpg" alt="">
</section>

.photo-page img {
    width: auto;
    height: auto;
    max-width: 96%;
    max-height: 28.8cm;
    margin: 2%;
}

img {
    vertical-align: middle;
    border-style: none;
}

.photo-page {
    text-align: center;
}

.page {
    font-size: 12px;
    font-weight: 400;
    line-height: normal;
    color: #000;
}

**Expected behavior**
Vertical photos taken by phone should appear vertically in pdf as well as in html page.

**Possible Solution**

**Comment by PhilterPaper** [comment]
Is this problem _only_ with "portrait" images taken with a phone camera? I just tried it with a cropped image from a DSLR and it worked fine (0.12.5 with patched qt, Windows 10). I had also added the normal HTML stuff around your code (html, head, style, body tags). Make sure your HTML file is properly formed before spending any more time on it -- browsers tend to be very lenient with partial HTML, but wkHTMLtoPDF might not be. 

Do these phone images display OK when uploaded to a computer and displayed in a picture viewer or editor (Photoshop, GIMP, etc.), or do you need to rotate them? I don't know if a camera puts some sort of orientation information in the image and always reports the size in landscape mode, but wkHTMLtoPDF would almost certainly ignore such an orientation flag.

By the way, for listing code (including HTML tags), use a backtick around code items (an `<i>HTML tag</i>`, or a triple backtick before and after a block of code.

**Comment by gazlag** [comment]
> _Is this problem only with "portrait" images taken with a phone camera?_

Yes the problem only happens with vertical photos taken from the phone. The problem goes away when I save and upload the photos taken from the phone back in Photoshop.


> Do these phone images display OK when uploaded to a computer and displayed in a picture viewer or editor (Photoshop, GIMP, etc.), or do you need to rotate them?

It looks fine in the computer and image viewer. There is no need to rotate the photo.


![photo](https://user-images.githubusercontent.com/8068619/95580860-b83b2280-0a40-11eb-93f0-9f96646b9633.jpg)

**Comment by PhilterPaper** [comment]
Well, I can see from the Exif information that (as I feared) the width is greater than the height (landscape orientation) and an _orientation_ tag is used to correct the display to portrait. Apparently WebKit (used by wkHTMLtoPDF to do most of the work) doesn't pay attention to the orientation tag, and outputs the raw image (without rotation). 

If that's the case, until WebKit is updated, it looks like you'll have to use a photo editor to rotate (possibly automatically) and save in portrait mode anything coming from a phone camera.

**Comment by fw6** [comment]
The style below can solve this problem! Try it!:monocle_face:
```css
img {
      image-orientation: from-image; /* Use EXIF data from the image */
}
```

**Comment by PhilterPaper** [comment]
Be careful using this solution. According to https://developer.mozilla.org/en-US/docs/Web/CSS/image-orientation, `image-orientation` is deprecated and may change or disappear! However, they don't seem to give any replacement for it. If it happens to work under the present WebKit, there's no guarantee that a future update won't change things.

**Comment by microt-mi** [comment]
I had the same problem and used `image-orientation: from-image; `it doesn't work.

**Comment by alarm109** [comment]
The solution i found: using [exif-orientation](https://github.com/ginpei/exif-orientation) determine if the image is flipped. If not, then add the rotation to the image through css.
